### PR TITLE
Massive ES6 Port

### PR DESCRIPTION
**Pull request by MarshallOfSound**
* New concept of Structs, currently only a Track
* Each namespace is it's own class
* Test's pass

### CODE REVIEW

File: GMusic 6 CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Changelog
```
Comment (jostrander) [inline GMusic 6 CHANGELOG.md]: Can we move this changelog to the general readme under the title 6.0.0?

Comment (MarshallOfSound) [inline GMusic 6 CHANGELOG.md]: Yeah, this was just a temp file so I could keep track of what I did 😆

**Comment by jostrander** [comment]
Can we get a review from @jacobwgillespie or @chrismou on this as well. I'd like to get this merged (pending the changelog and docs changes) and fix the podcasts stuff from #3.

**Comment by jacobwgillespie** [comment]
Thought I'd already +1'd this - sorry!

:shipit:

**Comment by MarshallOfSound** [comment]
Worth pointing out that with this PR we no longer keep our dist files in version control.

Instead we now have a multi loader like this.

```js
if (typeof global.define === 'function' && global.define.amd) {
  global.define('gmusic.js', [], () => GMusic);
} else if (typeof global.module === 'object' && module.exports) {
  module.exports = GMusic;
} else {
  window.GMusic = GMusic;
}
```

This allows people to browserify or just load our dist files from NPM and it will "Just Work" wherever they load it

**Comment by MarshallOfSound** [comment]
Can I get a Go / No Go from @jostrander @jacobwgillespie @chrismou 

Tests are green and changelog is updated 👍

**Comment by jostrander** [comment]
:shipit:

**Comment by chrismou** [comment]
:+1:

**Comment by jacobwgillespie** [comment]
:shipit:

**Pull request merged**